Use Java 15 text blocks for JSON and multiline strings

arteam opened the pull request:
The ES code base is quite JSON heavy. It uses a lot of multi-line JSON requests in tests which need to be escaped and concatenated which in turn makes them hard to read. Let's try to leverage Java 15 text blocks for representing them.

pugnascotia commented on client/rest-high-level/src/test/java/org/elasticsearch/client/BulkProcessorIT.java:
```diff
@@ -212,16 +212,15 @@ public void testBulkProcessorWaitOnClose() throws Exception {
+                        "blocks.write" : true
+                    }
+                }
+               \s
```
That looks peculiar - we could just remove this line?

arteam replied:
Yeah, the PR is still very WIP. I'm going to go over all the strings manually and fix peculiarities like that.

pugnascotia commented:
I'm happy to go over this properly when we're in a position to merge 👍

arteam committed:
Migrate escaped JSON requests to text blocks

ChrisHegarty reviewed (commented):
Generally, I think that this looks great.

One observation. A lot of text blocks have the closing delimited directly after the end of the text, which is fine but would be less than ideal if we ever want to add lead whitespace. Not sure if the use cases would ever need this, just checking that some consideration has been given to it, see https://openjdk.java.net/projects/amber/guides/text-blocks-guide#G12

arteam committed:
WIP: Use Java 15 text blocks for multiline strings
Remove supress:warnings

# Conflicts:
#	test/framework/src/test/java/org/elasticsearch/test/rest/yaml/restspec/test.json

arteam commented:
@elasticmachine run elasticsearch-ci/part-2

arteam commented:
@elasticmachine 

run elasticsearch-ci/eql-correctness
run elasticsearch-ci/part-2
run elasticsearch-ci/rest-compatibility

elasticmachine commented:
Pinging @elastic/es-delivery (Team:Delivery)

elasticmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

arteam commented:
@elasticmachine run elasticsearch-ci/bwc

nik9000 commented on build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/DependenciesGraphTask.java:
```diff
@@ -90,50 +90,49 @@ void generateDependenciesGraph() {
-                .append(dependency.getVersion())
-                .append("\"}}");
+            packageString.append("""
+                {"id": "%s","info": {"name": "%s","version": "%s"}}""".formatted(versionedId, id, dependency.getVersion()));
```
Do we need a locale here? The docs and the code seem to indicate that this uses the platform default locale. We'd forbid that in regular `String.format`

arteam replied:
`formatted` doesn't accept a locale :( But AFAIK all calls to formatted use `%s` for string interpolation, so they should be locale independent.

pugnascotia commented on build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/DependenciesGraphTask.java:
```diff
@@ -85,55 +85,55 @@ void generateDependenciesGraph() {
+                    }
+                  }
+                }""".formatted(
+                projectName,
```
It's not essential, but you could only supply `projectName` once here, and refer to it in the format string with e.g. `%1$s`

pugnascotia commented on build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerSupportService.java:
```diff
@@ -154,8 +154,11 @@ void failIfDockerUnavailable(List<String> tasks) {
-            "a problem occurred while using Docker from [%s]%s yet it is required to run the following task%s: \n%s\n"
-                + "the problem is that Docker exited with exit code [%d] with standard error output:\n%s",
+            """
+                a problem occurred while using Docker from [%s]%s yet it is required to run the following task%s:\s
```
Should this be:

```suggestion
                a problem occurred while using Docker from [%s]%s yet it is required to run the following task%s:
```

pugnascotia commented on build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/DependencyLicensesTask.java:
```diff
@@ -318,16 +318,10 @@ private void checkSha(File jar, String jarName, Set<File> shaFiles) throws NoSuc
+            final String exceptionMessage = String.format(Locale.ROOT, """
+                SHA has changed! Expected %s for %s but got %s.
+                This usually indicates a corrupt dependency cache or artifacts changed upstream.
+                Either wipe your cache, fix the upstream artifact, or delete %s and run updateShas""", expectedSha, jarName, sha, shaFile);
```
```suggestion
                Either wipe your cache, fix the upstream artifact, or delete %s and run updateShas
                """, expectedSha, jarName, sha, shaFile);
```

pugnascotia commented on client/rest-high-level/src/test/java/org/elasticsearch/client/IngestRequestConvertersTests.java:
```diff
@@ -82,29 +82,14 @@ public void testDeletePipeline() {
+                "description": "_description",
+                "processors": [ { "set": { "field": "field2", "value": "_value" } } ]
+              },
+              "docs": [ { "_index": "index", "_type": "_doc", "_id": "id", "_source": { "foo": "rab" } } ]
```
It looks like the original JSON was intended to be spread over more lines, except it didn't include the newlines. Maybe can put each key-pair on its own line, line the original?

pugnascotia reviewed (commented):
I saw the conversation around what style of terminating `"""` we wanted, and put together a command to convert to a trailing newline style:

```shell
git grep -l '[^ "]"""[;,)]' '**/*.java' | xargs -n1 perl -i -p -e 's/^(\s*)([^" ]+)"""([,;)].*)/ $1 . $2 . "\n" . (" " x length($1)) . qq(""") . $3 /e'
```

I don't know if it'll catch every case, but I hope it's useful.

pugnascotia commented on client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SnapshotClientDocumentationIT.java:
```diff
@@ -133,8 +133,9 @@ public void testSnapshotCreateRepository() throws IOException {
-            request.settings("{\"location\": \".\", \"compress\": \"true\"}",
-                XContentType.JSON); // <1>
+            request.settings("""
+                    {"location": ".", "compress": "true"}
+                    """, XContentType.JSON); // <1>
```
```suggestion
                {"location": ".", "compress": "true"}
                """, XContentType.JSON); // <1>
```

pugnascotia commented on client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/WatcherDocumentationIT.java:
```diff
@@ -144,11 +144,12 @@ public void testWatcher() throws Exception {
-                "  \"actions\": { \"logme\": { \"logging\": { \"text\": \"{{ctx.payload}}\" } } }\n" +
-                "}");
+            BytesReference watch = new BytesArray("""
+                    {
+                      "trigger": { "schedule": { "interval": "10h" } },
+                      "input": { "simple": { "foo" : "bar" } },
+                      "actions": { "logme": { "logging": { "text": "{{ctx.payload}}" } } }
+                    }""");
```
```suggestion
                {
                  "trigger": { "schedule": { "interval": "10h" } },
                  "input": { "simple": { "foo" : "bar" } },
                  "actions": { "logme": { "logging": { "text": "{{ctx.payload}}" } } }
                }""");
```

pugnascotia commented on client/rest-high-level/src/test/java/org/elasticsearch/client/security/ClearRealmCacheResponseTests.java:
```diff
@@ -54,7 +53,8 @@ public void testParseFromXContent() throws IOException {
                 NamedXContentRegistry.EMPTY,
                 DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
-                "{" + nodesHeader + clusterName + "\"nodes\" : { \"id1\": { \"name\": \"a\"}, \"id2\": { \"name\": \"b\"}}}"
+                "{" + nodesHeader + clusterName + """
+                    "nodes" : { "id1": { "name": "a"}, "id2": { "name": "b"}}}"""
```
If we took out the trailing comma from `nodesHeader` and `clusterName`, then we could do something like:
```suggestion
                """
                    {
                      %s,
                      %s,
                      "nodes" : { "id1": { "name": "a"}, "id2": { "name": "b"}}
                    }
                    """.formatted(nodesHeader, clusterName)
```

pugnascotia commented on client/rest-high-level/src/test/java/org/elasticsearch/client/security/ClearRolesCacheResponseTests.java:
```diff
@@ -54,7 +54,9 @@ public void testParseFromXContent() throws IOException {
                 DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
-                "{" + nodesHeader + clusterName + "\"nodes\" : { \"id1\": { \"name\": \"a\"}, \"id2\": { \"name\": \"b\"}}}"
+                """
+                    {%s%s"nodes" : { "id1": { "name": "a"}, "id2": { "name": "b"}}}
```
Same here, we can take out the trailing comma from `nodesHeader` and `clusterName` and make this more readable.

pugnascotia commented on client/rest-high-level/src/test/java/org/elasticsearch/client/security/GrantApiKeyRequestTests.java:
```diff
@@ -42,19 +43,18 @@ public void testToXContent() throws IOException {
-                    + "}"
-            )
-        );
+        assertThat(output, equalTo(XContentHelper.stripWhitespace("""
```
I knew this method would come in handy 😄

pugnascotia commented on libs/x-content/src/test/java/org/elasticsearch/xcontent/DotExpandingXContentParserTests.java:
```diff
@@ -26,27 +26,24 @@ private void assertXContentMatches(String expected, String actual) throws IOExce
-            "{\"test.with.dots\":{\"field\":\"value\"},\"nodots\":\"value2\"}"
-        );
+        assertXContentMatches("""
+            {"test":{"with":{"dots":{"field":"value"}}},"nodots":"value2"}""", """
```
My shell command doesn't catch these, BTW.

pugnascotia commented on modules/lang-mustache/src/test/java/org/elasticsearch/script/mustache/RestMultiSearchTemplateActionTests.java:
```diff
@@ -35,7 +35,10 @@ public void setUpAction() {
     public void testTypeInPath() {
-        String content = "{ \"index\": \"some_index\" } \n" + "{\"source\": {\"query\" : {\"match_all\" :{}}}} \n";
+        String content = """
+            { "index": "some_index" }\s
+            {"source": {"query" : {"match_all" :{}}}}\s
```
```suggestion
            { "index": "some_index" }
            {"source": {"query" : {"match_all" :{}}}}
```

pugnascotia commented on modules/lang-mustache/src/test/java/org/elasticsearch/script/mustache/RestMultiSearchTemplateActionTests.java:
```diff
@@ -47,7 +50,10 @@ public void testTypeInPath() {
     public void testTypeInBody() {
-        String content = "{ \"index\": \"some_index\", \"type\": \"some_type\" } \n" + "{\"source\": {\"query\" : {\"match_all\" :{}}}} \n";
+        String content = """
+            { "index": "some_index", "type": "some_type" }\s
+            {"source": {"query" : {"match_all" :{}}}}\s
```
```suggestion
            { "index": "some_index", "type": "some_type" }
            {"source": {"query" : {"match_all" :{}}}}
```

pugnascotia commented on modules/lang-mustache/src/test/java/org/elasticsearch/script/mustache/SearchTemplateRequestXContentTests.java:
```diff
@@ -125,41 +128,51 @@ public void testToXContentWithStoredTemplate() throws IOException {
-            + "}";
+        String source = """
+            {
+              'source' : {
```
What do you think - is it worth removing the `s.replace("'", "\"")` in `newParser()` and using standard double quotes in the JSON, here and. below?

arteam committed:
Run spotless

pugnascotia commented on plugins/analysis-icu/src/internalClusterTest/java/org/elasticsearch/index/mapper/ICUCollationKeywordFieldMapperIT.java:
```diff
@@ -306,12 +306,13 @@ public void testIgnoreWhitespace() throws Exception {
-            client().prepareIndex(index).setId("3").setSource("{\"id\":\"3\",\"collate\":\"foo-bar\"}", XContentType.JSON)
-        );
+        indexRandom(true, client().prepareIndex(index).setId("1").setSource("""
+            {"id":"1","collate":"foo bar"}
```
I don't think this particular file is more readable for these changes. What do you think about keeping the old version?

pugnascotia commented on qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java:
```diff
@@ -569,26 +569,31 @@ public void testShrinkAfterUpgrade() throws IOException {
+        String bulk = """
+            {"index":{}}
+            {"test":"test"}
+            """.repeat(bulkCount);
```
🎉 👏

pugnascotia commented on qa/logging-config/src/test/java/org/elasticsearch/common/logging/ESJsonLayoutTests.java:
```diff
@@ -25,45 +27,21 @@ public void testLayout() {
-            )
-        );
+        assertThat(conversionPattern, Matchers.equalTo(String.format(Locale.ROOT, """
+            {"type": "server", "timestamp": "%%d{yyyy-MM-dd'T'HH:mm:ss,SSSZZ}", "level": "%%p", "component": "%%c{1.}", \
```
Can we split this over multiple lines, to make it more readable? Same thing below as well.

pugnascotia commented on qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/SearchWithMinCompatibleSearchNodeIT.java:
```diff
@@ -84,19 +84,11 @@ public void testMinVersionAsNewVersion() throws Exception {
-                    )
-                );
+                assertThat(responseException.getMessage(), containsString("""
+                    {"error":{"root_cause":[],"type":"search_phase_execution_exception\""""));
```
```suggestion
                    {"error":{"root_cause":[],"type":"search_phase_execution_exception"
                    """));
```

pugnascotia commented on qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/SearchWithMinCompatibleSearchNodeIT.java:
```diff
@@ -84,19 +84,11 @@ public void testMinVersionAsNewVersion() throws Exception {
+                    {"error":{"root_cause":[],"type":"search_phase_execution_exception\""""));
+                assertThat(responseException.getMessage(), containsString("""
+                    caused_by":{"type":"version_mismatch_exception",\
+                    "reason":"One of the shards is incompatible with the required minimum version [%s]\"""".formatted(newVersion)));
```
```suggestion
                    "reason":"One of the shards is incompatible with the required minimum version [%s]"
                    """.formatted(newVersion)));
```

pugnascotia commented on qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/SearchWithMinCompatibleSearchNodeIT.java:
```diff
@@ -143,10 +136,8 @@ public void testCcsMinimizeRoundtripsIsFalse() throws Exception {
-                        containsString("{\"error\":{\"root_cause\":[{\"type\":\"action_request_validation_exception\"")
-                    );
+                    assertThat(responseException.getMessage(), containsString("""
+                        {"error":{"root_cause":[{"type":"action_request_validation_exception\""""));
```
```suggestion
                        {"error":{"root_cause":[{"type":"action_request_validation_exception"
                        """));
```

pugnascotia commented on qa/os/src/test/java/org/elasticsearch/packaging/test/PackagesSecurityAutoConfigurationTests.java:
```diff
@@ -154,10 +154,10 @@ public void test71ReconfigureFailsWhenKeyStorePasswordWrong() throws Exception {
-            "passwd",
-            "some-password" + "\n" + "some-password" + "\n"
-        );
+        Shell.Result changePassword = installation.executables().keystoreTool.run("passwd", """
+            some-password
+            some-password
+            """);
```
Given how short the second string is, I personally think this code would be more readable as:
```suggestion
        Shell.Result changePassword = installation.executables().keystoreTool.run("passwd", "some-password\nsome-password\n");
```

pugnascotia commented on server/src/internalClusterTest/java/org/elasticsearch/get/GetActionIT.java:
```diff
@@ -781,37 +789,33 @@ public void testGeneratedStringFieldsStored() throws IOException {
+              },
+              "mappings": {
+                "_doc": {
+                  "_source" : {"enabled" : %s},      "properties": {
```
This JSON is a bit mangled.

pugnascotia commented on server/src/internalClusterTest/java/org/elasticsearch/indices/template/SimpleIndexTemplateIT.java:
```diff
@@ -607,24 +599,18 @@ public void testIndexTemplateWithAliasesInSource() {
-            )
-            .get();
+        client().admin().indices().preparePutTemplate("template_1").setPatterns(Collections.singletonList("te*")).setAliases("""
+                {
+                    "alias1" : {},
+                    "alias2" : {
+                        "filter" : {
+                            "term" : {
+                                "field" : "value2"
+                            }
+                        }
+                     },
+                    "alias3" : { "routing" : "1" }    }
```
```suggestion
            {
                "alias1" : {},
                "alias2" : {
                    "filter" : {
                        "term" : {
                            "field" : "value2"
                        }
                    }
                 },
                "alias3" : { "routing" : "1" }
            }
```

pugnascotia commented on server/src/internalClusterTest/java/org/elasticsearch/indices/template/SimpleIndexTemplateIT.java:
```diff
@@ -830,25 +816,19 @@ public void testCombineTemplates() throws Exception {
-            )
-            .get();
+        client().admin().indices().preparePutTemplate("template_1").setPatterns(Collections.singletonList("*")).setSettings("""
+                {
+                    "index" : {
+                        "analysis" : {
+                            "analyzer" : {
+                                "custom_1" : {
+                                    "tokenizer" : "standard"
+                                }
+                            }
+                        }
+                     }
+                }
```
```suggestion
            {
                "index" : {
                    "analysis" : {
                        "analyzer" : {
                            "custom_1" : {
                                "tokenizer" : "standard"
                            }
                        }
                    }
                 }
            }
```

pugnascotia replied:
```suggestion
            {
                "index" : {
                    "analysis" : {
                        "analyzer" : {
                            "custom_1" : {
                                "tokenizer" : "standard"
                            }
                        }
                    }
                 }
            }
```

pugnascotia commented on server/src/internalClusterTest/java/org/elasticsearch/search/nested/SimpleNestedIT.java:
```diff
@@ -929,71 +916,58 @@ public void testNestedSortWithMultiLevelFiltering() throws Exception {
-                )
-        );
+        assertAcked(prepareCreate("test").setSettings(Settings.builder().put("number_of_shards", 1)).setMapping("""
+            {"_doc":{
+                    "dynamic": "strict",
+                    "properties": {
+                      "nested1": {
+                        "type": "nested",
+                        "properties": {
+                          "nested2": {
+                            "type": "nested",
+                            "properties": {
+                              "nested2_keyword": {
+                                "type": "keyword"
+                              },
+                              "sortVal": {
+                                "type": "integer"
+                              }
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }}
```
```suggestion
            {
              "_doc": {
                "dynamic": "strict",
                "properties": {
                  "nested1": {
                    "type": "nested",
                    "properties": {
                      "nested2": {
                        "type": "nested",
                        "properties": {
                          "nested2_keyword": {
                            "type": "keyword"
                          },
                          "sortVal": {
                            "type": "integer"
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
```

pugnascotia commented on server/src/internalClusterTest/java/org/elasticsearch/search/nested/SimpleNestedIT.java:
```diff
@@ -929,71 +916,58 @@ public void testNestedSortWithMultiLevelFiltering() throws Exception {
+                      "sortVal": 2
+                    }
+                  ]
+                }\s
```
```suggestion
                }
```

pugnascotia commented on server/src/main/java/org/elasticsearch/bootstrap/JNANatives.java:
```diff
@@ -92,15 +92,11 @@ static void tryMlockall() {
-                        user
-                    );
+                    logger.warn("""
+                        These can be adjusted by modifying /etc/security/limits.conf, for example:\s
```
```suggestion
                        These can be adjusted by modifying /etc/security/limits.conf, for example:
```

pugnascotia commented on server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java:
```diff
@@ -636,25 +636,15 @@ private static String normalizeValue(NamedAnalyzer normalizer, String field, Str
-                );
+                throw new IllegalStateException("""
+                    The normalization token stream is expected to produce exactly 1 token, \
+                    but got 0 for analyzer %s and input "%s\"""".formatted(normalizer, value));
```
```suggestion
                    but got 0 for analyzer %s and input "%s"
                    """.formatted(normalizer, value));
```

pugnascotia commented on server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java:
```diff
@@ -636,25 +636,15 @@ private static String normalizeValue(NamedAnalyzer normalizer, String field, Str
-                );
+                throw new IllegalStateException("""
+                    The normalization token stream is expected to produce exactly 1 token, \
+                    but got 2+ for analyzer %s and input "%s\"""".formatted(normalizer, value));
```
```suggestion
                    but got 2+ for analyzer %s and input "%s"
                    """.formatted(normalizer, value));
```

pugnascotia commented on server/src/main/java/org/elasticsearch/index/shard/RemoveCorruptedShardDataCommand.java:
```diff
@@ -197,13 +197,12 @@ public static boolean isCorruptMarkerFileIsPresent(final Directory directory) th
-            );
+            confirm("""
+                This shard has been marked as corrupted but no corruption can now be detected.
+                This may indicate an intermittent hardware problem. The corruption marker can be\s
```
```suggestion
                This may indicate an intermittent hardware problem. The corruption marker can be
```

pugnascotia commented on server/src/main/java/org/elasticsearch/indices/SystemIndices.java:
```diff
@@ -643,13 +643,9 @@ private static boolean overlaps(String pattern1, String pattern2) {
-                    + plugin
-                    + "]"
-            );
+            throw new IllegalArgumentException("""
+                feature name cannot be reserved name ["%s"], but was for plugin [%s]\
+                """.formatted(SnapshotsService.NO_FEATURE_STATES_VALUE, plugin));
```
I think this would be easier more legible:
```suggestion
            throw new IllegalArgumentException(
                "feature name cannot be reserved name [\"%s\"], but was for plugin [%s]".formatted(
                    SnapshotsService.NO_FEATURE_STATES_VALUE,
                    plugin
                )
            );
```

pugnascotia commented on server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/heuristic/NXYSignificanceHeuristic.java:
```diff
@@ -26,10 +26,10 @@
-        + BACKGROUND_IS_SUPERSET.getPreferredName()
-        + "\": false";
+    protected static final String SCORE_ERROR_MESSAGE = """
+        , does your background filter not include all documents in the bucket? If so and it is intentional, set "%s": false""".formatted(
+        BACKGROUND_IS_SUPERSET.getPreferredName()
+    );
```
```suggestion
        , does your background filter not include all documents in the bucket? If so and it is intentional, set "%s": false
        """.formatted(BACKGROUND_IS_SUPERSET.getPreferredName());
```

arteam committed:
Fix the length of the json input

pugnascotia commented on server/src/main/java/org/elasticsearch/transport/RemoteConnectionStrategy.java:
```diff
@@ -430,17 +430,10 @@ public void validate(T value, Map<Setting<?>, Object> settings, boolean isPresen
-                        + modeType.name()
-                        + "]"
-                );
+                throw new IllegalArgumentException("""
+                    Setting "%s" cannot be used with the configured "%s" \
+                    [required=%s, configured=%s]\
+                    """.formatted(key, concrete.getKey(), expectedStrategy.name(), modeType.name()));
```
What about:

```suggestion
                throw new IllegalArgumentException(
                    "Setting \"%s\" cannot be used with the configured \"%s\" [required=%s, configured=%s]".formatted(
                        key,
                        concrete.getKey(),
                        expectedStrategy.name(),
                        modeType.name()
                    )
                );
```

pugnascotia commented on server/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequestBuilderTests.java:
```diff
@@ -50,13 +50,14 @@ public void tearDown() throws Exception {
-            ElasticsearchParseException.class,
-            () -> { builder.setSource("{\"" + KEY + "\" : \"" + VALUE + "\"}", XContentType.JSON); }
-        );
+        ElasticsearchParseException e = expectThrows(ElasticsearchParseException.class, () -> builder.setSource("""
+            {"%s" : "%s"}
+            """.formatted(KEY, VALUE), XContentType.JSON));
```
How about:
```suggestion
        ElasticsearchParseException e = expectThrows(
            ElasticsearchParseException.class,
            () -> builder.setSource("{ \"%s\": \"%s\" }".formatted(KEY, VALUE), XContentType.JSON)
        );
```

arteam committed:
Update server/src/main/java/org/elasticsearch/transport/RemoteConnectionStrategy.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequestBuilderTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/main/java/org/elasticsearch/indices/SystemIndices.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/main/java/org/elasticsearch/index/shard/RemoveCorruptedShardDataCommand.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

pugnascotia commented on server/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequestBuilderTests.java:
```diff
@@ -50,13 +50,14 @@ public void tearDown() throws Exception {
         assertEquals(String.format(Locale.ROOT, "unknown key [%s] for create index", KEY), e.getMessage());
 
-        builder.setSource("{\"settings\" : {\"" + KEY + "\" : \"" + VALUE + "\"}}", XContentType.JSON);
+        builder.setSource("""
+            {"settings" : {"%s" : "%s"}}
+            """.formatted(KEY, VALUE), XContentType.JSON);
```
How about:
```suggestion
        builder.setSource("{ \"settings\": { \"%s\": \"%s\" }}".formatted(KEY, VALUE), XContentType.JSON);
```

arteam committed:
Update server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/heuristic/NXYSignificanceHeuristic.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/main/java/org/elasticsearch/bootstrap/JNANatives.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/internalClusterTest/java/org/elasticsearch/search/nested/SimpleNestedIT.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/internalClusterTest/java/org/elasticsearch/search/nested/SimpleNestedIT.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequestBuilderTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/internalClusterTest/java/org/elasticsearch/indices/template/SimpleIndexTemplateIT.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/internalClusterTest/java/org/elasticsearch/indices/template/SimpleIndexTemplateIT.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update qa/os/src/test/java/org/elasticsearch/packaging/test/PackagesSecurityAutoConfigurationTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/SearchWithMinCompatibleSearchNodeIT.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

pugnascotia commented on server/src/test/java/org/elasticsearch/action/delete/DeleteResponseTests.java:
```diff
@@ -29,26 +30,39 @@
-            );
+            assertEquals(XContentHelper.stripWhitespace("""
+                {
+                          "_index": "index",
+                          "_id": "id",
+                          "_version": 7,
+                          "result": "deleted",
+                          "forced_refresh": true,
+                          "_shards": {
+                            "total": 10,
+                            "successful": 5,
+                            "failed": 0
+                          }
+                        }"""), output);
```
```suggestion
                  "_index": "index",
                  "_id": "id",
                  "_version": 7,
                  "result": "deleted",
                  "forced_refresh": true,
                  "_shards": {
                    "total": 10,
                    "successful": 5,
                    "failed": 0
                  }
                }
                """), output);
```

pugnascotia commented on server/src/test/java/org/elasticsearch/action/explain/ExplainResponseTests.java:
```diff
@@ -94,30 +95,29 @@ public void testToXContent() throws IOException {
-            + "}").replaceAll("\\s+", "");
+        String expectedResponse = ("""
+            {
+                "_index":"index",
+                "_id":"1",
+                "matched":true,
+                "explanation":{
+                    "value":1.0,
+                    "description":"description",
+                    "details":[]
+                },
+                "get":{
+                    "_seq_no":0,        "_primary_term":1,        "found":true,
+                    "_source":{
+                        "field1":"value1",
+                        "field2":"value2"
+                    },
+                    "fields":{
+                        "field1":[
+                            "value1"
+                        ]
+                    }
+                }
+            }""").replaceAll("\\s+", "");
```
```suggestion
                "_index": "index",
                "_id": "1",
                "matched": true,
                "explanation": {
                    "value": 1.0,
                    "description": "description",
                    "details": []
                },
                "get": {
                    "_seq_no": 0,
                    "_primary_term": 1,
                    "found": true,
                    "_source": {
                        "field1": "value1",
                        "field2": "value2"
                    },
                    "fields": {
                        "field1": [
                          "value1"
                        ]
                    }
                }
            }
            """).replaceAll("\\s+", "");
```

pugnascotia replied:
```suggestion
                "_index": "index",
                "_id": "1",
                "matched": true,
                "explanation": {
                    "value": 1.0,
                    "description": "description",
                    "details": []
                },
                "get": {
                    "_seq_no": 0,
                    "_primary_term": 1,
                    "found": true,
                    "_source": {
                        "field1": "value1",
                        "field2": "value2"
                    },
                    "fields": {
                        "field1": [
                          "value1"
                        ]
                    }
                }
            }
            """).replaceAll("\\s+", "");
```

pugnascotia commented on server/src/test/java/org/elasticsearch/action/index/IndexRequestTests.java:
```diff
@@ -139,23 +139,19 @@ public void testIndexResponse() {
-                + successful
-                + ",\"failed\":0}]",
+            """
+                IndexResponse[index=%s,id=%s,version=%s,result=%s,seqNo=%s,primaryTerm=%s,shards=\
```
Do we really want to have an assertion that the `toString()` output is as we expect? That seems a bit brittle?

pugnascotia commented on server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java:
```diff
@@ -1807,45 +1658,27 @@ public void testUpdateComponentTemplateFailsIfResolvedIndexTemplatesWouldBeInval
+                }"""), null), null, null);
+        ComponentTemplate ct2 = new ComponentTemplate(new Template(null, new CompressedXContent("""
+            {
+                  "properties": {
+                    "field1": {
+                      "type": "text"
+                    }
+                  }
+                }"""), null), null, null);
```
```suggestion
              "properties": {
                "field1": {
                  "type": "text"
                }
              }
            }
            """), null), null, null);
```

pugnascotia commented on server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java:
```diff
@@ -1862,23 +1695,14 @@ public void testUpdateComponentTemplateFailsIfResolvedIndexTemplatesWouldBeInval
-        );
+        ComponentTemplate changedCt2 = new ComponentTemplate(new Template(null, new CompressedXContent("""
+            {
+                  "properties": {
+                    "field2": {
+                      "type": "text"
+                    }
+                  }
+                }"""), null), null, null);
```
```suggestion
              "properties": {
                "field2": {
                  "type": "text"
                }
              }
            }
            """), null), null, null);
```

pugnascotia commented on server/src/test/java/org/elasticsearch/common/logging/JsonThrowablePatternConverterTests.java:
```diff
@@ -37,25 +37,22 @@ public void testNoStacktrace() throws IOException {
-            + "  }"
-            + LINE_SEPARATOR
-            + "}";
+        String json = """
+            {%s  \
+            "terms" : {%s \
+            "user" : [%s "u1",%s "u2",%s "u3"%s ], %s \
+            "boost" : 1.0%s  \
+            }%s}""".formatted(
+            LINE_SEPARATOR,
+            LINE_SEPARATOR,
+            LINE_SEPARATOR,
+            LINE_SEPARATOR,
+            LINE_SEPARATOR,
+            LINE_SEPARATOR,
+            LINE_SEPARATOR,
+            LINE_SEPARATOR,
+            LINE_SEPARATOR
+        );
```
The original code is bizarre. We can make it simpler:

```suggestion
        String json = """
            {
              "terms": {
                "user": [
                  "u1",
                  "u2",
                  "u3"
                ],
                "boost": 1.0
              }
            }
            """;
```

arteam committed:
Update server/src/test/java/org/elasticsearch/action/delete/DeleteResponseTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/action/explain/ExplainResponseTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/common/logging/JsonThrowablePatternConverterTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

pugnascotia commented on server/src/test/java/org/elasticsearch/index/mapper/DynamicTemplatesTests.java:
```diff
@@ -685,16 +682,8 @@ public void testTemplateWithoutMatchPredicates() throws Exception {
-                    Map.of("foo", "geo_point")
-                )
-            );
+        ParsedDocument doc = mapperService.documentMapper().parse(new SourceToParse("1", new BytesArray("""
+            {"foo": "41.12,-71.34", "bar": "41.12,-71.34"}"""), XContentType.JSON, null, Map.of("foo", "geo_point")));
```
In my opinion, this is a case where it's clearer to extract the JSON:
```suggestion
        final String json = """
            {"foo": "41.12,-71.34", "bar": "41.12,-71.34"}
            """;
        ParsedDocument doc = mapperService.documentMapper()
            .parse(new SourceToParse("1", new BytesArray(json), XContentType.JSON, null, Map.of("foo", "geo_point")));
```

arteam committed:
Update server/src/test/java/org/elasticsearch/index/mapper/DynamicTemplatesTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

pugnascotia commented on server/src/test/java/org/elasticsearch/index/mapper/FieldFilterMapperPluginTests.java:
```diff
@@ -280,71 +280,68 @@ private static void assertNotFiltered(MappingMetadata mappingMetadata) {
+                      "type": "geo_point"
+                    },
+                    "area_visible": {
+                      "type": "geo_shape"            }
```
```suggestion
                      "type": "geo_shape"
                    }
```

arteam committed:
Update server/src/test/java/org/elasticsearch/index/mapper/FieldFilterMapperPluginTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

pugnascotia commented on server/src/test/java/org/elasticsearch/index/mapper/ParametrizedMapperTests.java:
```diff
@@ -296,116 +296,165 @@ private String toStringWithDefaults(ToXContent value) throws IOException {
-            "{\"field\":{\"type\":\"test_mapper\",\"fixed\":false,\"variable\":\"updated\",\"required\":\"value\"}}",
-            Strings.toString(merged)
-        );
+        assertEquals("""
+            {"field":%s}\
+            """.formatted(mapping), Strings.toString(mapper));   // original mapping is unaffected
```
I think we should leave this line alone, the same line also appears above on line 344.
```suggestion
assertEquals("{\"field\":" + mapping + "}", Strings.toString(mapper)); // original mapping is unaffected
```

pugnascotia commented on server/src/test/java/org/elasticsearch/index/mapper/ParametrizedMapperTests.java:
```diff
@@ -296,116 +296,165 @@ private String toStringWithDefaults(ToXContent value) throws IOException {
+            {"type":"test_mapper","variable":null,"required":"value"}""";
         TestMapper mapper = fromMapping(fine);
-        assertEquals("{\"field\":" + fine + "}", Strings.toString(mapper));
+        assertEquals("""
+            {"field":%s}\
+            """.formatted(fine), Strings.toString(mapper));
```
Again, I think it's not worth changing:
```suggestion
assertEquals("{\"field\":" + fine + "}", Strings.toString(mapper));
```

arteam committed:
Update server/src/test/java/org/elasticsearch/index/mapper/ParametrizedMapperTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/index/mapper/ParametrizedMapperTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam commented:
@elasticmachine update branch

arteam committed:
Run spotless

arteam commented:
@elasticmachine update branch

pugnascotia commented on server/src/test/java/org/elasticsearch/index/query/PrefixQueryBuilderTests.java:
```diff
@@ -96,10 +93,10 @@ public void testBlendedRewriteMethod() throws IOException {
-            + " \"boost\" : 2.0"
-            + "} }}";
+        String json = """
+            {    "prefix" : { "user" :  { "value" : "ki",
+                 "case_insensitive" : true,
+             "boost" : 2.0} }}""";
```
```suggestion
            {
              "prefix": {
                "user": {
                  "value": "ki",
                  "case_insensitive": true,
                  "boost": 2.0
                }
              }
            }
            """;
```

pugnascotia commented on server/src/test/java/org/elasticsearch/index/query/ScriptQueryBuilderTests.java:
```diff
@@ -67,31 +68,32 @@ public void testFromJsonVerbose() throws IOException {
+        String json = """
+            {
+              "script" : {
+                "script" : "5",    "boost" : 1.0,
```
```suggestion
                "script" : "5",
                "boost" : 1.0,
```

pugnascotia commented on server/src/test/java/org/elasticsearch/index/query/ScriptScoreQueryBuilderTests.java:
```diff
@@ -59,15 +59,16 @@ protected void doAssertLuceneQuery(ScriptScoreQueryBuilder queryBuilder, Query q
+              "script_score" : {
+                "query" : { "match_all" : {} },
+                "script" : {
+                  "source" : "doc['field'].value"\s
```
```suggestion
                  "source" : "doc['field'].value"
```

pugnascotia commented on server/src/test/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilderTests.java:
```diff
@@ -761,44 +772,55 @@ public void testSingleScriptFunction() throws IOException {
+                "function_score":{
+                    "query":{
+                        "bool":{
+                            "must":{"match":{"field":"value"}}             },
```
```suggestion
                            "must":{"match":{"field":"value"}}
                        },
```

pugnascotia commented on server/src/test/java/org/elasticsearch/index/search/MultiMatchQueryParserTests.java:
```diff
@@ -65,28 +65,25 @@ public void setup() throws IOException {
-            + "    }\n"
-            + "}";
+        String mapping = """
+            {
+                "person":{
+                    "properties":{
+                        "name":{
+                              "properties":{
+                                    "first": {
+                                        "type":"text",
+                                        "analyzer":"standard"
+                                    },                        "last": {
+                                        "type":"text",
+                                        "analyzer":"standard"
+                                    },                        "nickname": {
+                                        "type":"text",
+                                        "analyzer":"whitespace"
+                                    }                   }            }
+                    }
+                }
+            }""";
```
```suggestion
            {
                "person": {
                    "properties": {
                        "name": {
                            "properties": {
                                "first": {
                                    "type": "text",
                                    "analyzer": "standard"
                                },
                                "last": {
                                    "type": "text",
                                    "analyzer": "standard"
                                },
                                "nickname": {
                                    "type": "text",
                                    "analyzer": "whitespace"
                                }
                            }
                        }
                    }
                }
            }
            """;
```

pugnascotia commented on server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/SignificantTermsAggregatorTests.java:
```diff
@@ -589,7 +589,8 @@ private void addMixedTextDocs(IndexWriter w) throws IOException {
 
             doc.add(new Field("text", text.toString(), TextFieldMapper.Defaults.FIELD_TYPE));
-            String json = "{ \"text\" : \"" + text.toString() + "\" }";
+            String json = """
+                { "text" : "%s" }""".formatted(text.toString());
```
```suggestion
            String json = "{ \"text\" : \"%s\" }".formatted(text.toString());
```

pugnascotia commented on server/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java:
```diff
@@ -254,33 +259,33 @@ public void testParseSort() throws IOException {
-                + "}\n";
+            String restContent = """
+                {
+                    "aggs": {        "test_agg": {
```
```suggestion
                    "aggs": {
                        "test_agg": {
```

pugnascotia commented on server/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java:
```diff
@@ -254,33 +259,33 @@ public void testParseSort() throws IOException {
-                + "}\n";
+            String restContent = """
+                {
+                    "aggregations": {        "test_agg": {
```
```suggestion
                    "aggregations": {
                        "test_agg": {
```

pugnascotia commented on server/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java:
```diff
@@ -294,19 +299,20 @@ public void testAggsParsing() throws IOException {
+                    "query" : {
+                        "match": { "content": { "query": "foo bar" }}
+                     },
+                    "rescore": {        "window_size": 50,
```
```suggestion
                    "rescore": {
                        "window_size": 50,
```

pugnascotia commented on server/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java:
```diff
@@ -319,19 +325,20 @@ public void testParseRescore() throws IOException {
+                    "query" : {
+                        "match": { "content": { "query": "foo bar" }}
+                     },
+                    "rescore": [ {        "window_size": 50,
```
```suggestion
                    "rescore": [ {
                        "window_size": 50,
```

pugnascotia commented on server/src/test/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightBuilderTests.java:
```diff
@@ -213,24 +213,25 @@ public void testUnknownFieldnameExpection() throws IOException {
+            XContentParseException e = expectParseThrows(XContentParseException.class, """
+                {
+                    "bad_fieldname" :  { "field" : "value" }
+                \s
+                }
```
```suggestion
                }
```

pugnascotia commented on server/src/test/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightBuilderTests.java:
```diff
@@ -250,7 +251,12 @@ public void testStringInFieldsArray() throws IOException {
-        XContentParseException e = expectParseThrows(XContentParseException.class, "{\n" + "  \"fields\" : [ {\n" + "   }] \n" + "}\n");
+        XContentParseException e = expectParseThrows(XContentParseException.class, """
+            {
+              "fields" : [ {
+               }]\s
```
```suggestion
               }]
```

pugnascotia replied:
```suggestion
              "fields" : [ { } ]
```

arteam committed:
Update server/src/test/java/org/elasticsearch/index/query/PrefixQueryBuilderTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilderTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/index/query/ScriptScoreQueryBuilderTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/index/search/MultiMatchQueryParserTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightBuilderTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update server/src/test/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightBuilderTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam commented:
@elasticmachine update branch

elasticmachine commented:
merge conflict between base and head

arteam committed:
Update build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerSupportService.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/DependencyLicensesTask.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Fix tests

arteam committed:
Update client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SnapshotClientDocumentationIT.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update modules/lang-mustache/src/test/java/org/elasticsearch/script/mustache/RestMultiSearchTemplateActionTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update client/rest-high-level/src/test/java/org/elasticsearch/client/security/ClearRealmCacheResponseTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/WatcherDocumentationIT.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update minor formatting things

arteam committed:
Format more things

arteam committed:
Fix broken tests

arteam committed:
Update x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ExplainDataFrameAnalyticsIT.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/PyTorchModelIT.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/RegressionIT.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/IndexPrivilegeIntegTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/DiscountedCumulativeGainTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam commented on server/src/test/java/org/elasticsearch/common/logging/JsonThrowablePatternConverterTests.java:
```diff
@@ -67,7 +59,7 @@ public void testStacktraceWithJson() throws IOException {
             .orElseThrow(() -> new AssertionError("no logs parsed"));
 
-        int jsonLength = json.split(LINE_SEPARATOR).length;
+        int jsonLength = json.split("\\R").length;
```
I added a constant `NEWLINE` to make clear what `\|R` stands for.

arteam committed:
Extract \\R to a constant

arteam committed:
Update x-pack/plugin/sql/qa/server/src/main/java/org/elasticsearch/xpack/sql/qa/geo/GeoDataLoader.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam committed:
Update x-pack/plugin/text-structure/src/test/java/org/elasticsearch/xpack/textstructure/structurefinder/GrokPatternCreatorTests.java

Co-authored-by: Rory Hunter <pugnascotia@users.noreply.github.com>

arteam commented on x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/stats/VerifierMetricsTests.java:
```diff
@@ -52,19 +52,21 @@ public void testEventQuery() {
     public void testSequenceQuery() {
-        Counters c = eql("sequence\r\n" + "  [process where serial_event_id == 1]\r\n" + "  [process where serial_event_id == 2]");
+        Counters c = eql("""
+            sequence\r
```
Let me check that!

arteam committed:
Reformat rebased changes

arteam committed:
Run spotless

pugnascotia reviewed (approved):
LGTM 👍 thanks for taking on this work! 🚀

arteam commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into use-text-blocks

arteam committed:
Add missed import

arteam committed:
Remove comma

arteam commented:
Thanks, Rory!
